lib/github_changelog_generator/generator/entry.rb

```diff
@@ -0,0 +1,309 @@
+require "github_changelog_generator/generator/section"
+
+module GitHubChangelogGenerator
+  class Entry
+    attr_reader :content
+
+    def initialize(options = {})
+      @content = ""
+      @options = options
+    end
+
+    # Generates log entry with header and body
+    #
+    # @param [Array] pull_requests List or PR's in new section
+    # @param [Array] issues List of issues in new section
+    # @param [String] newer_tag Name of the newer tag. Could be nil for `Unreleased` section
+    # @param [Hash, nil] older_tag Older tag, used for the links. Could be nil for last tag.
+    # @return [String] Ready and parsed section
+    def create_entry_for_tag(pull_requests, issues, newer_tag, older_tag = nil)
+      newer_tag_link, newer_tag_name, newer_tag_time = detect_link_tag_time(newer_tag)
```

Comment:
Should an entry be told what the tags link/name/time is, or should it be able to figure it out just given a tag?

If the former, this logic moves up to the generator. If the latter, the methods for tag munging need to move into the entry class.

Reply:
The latter would be the better design decision, but `get_time_of_tag` is called by `detect_link_tag_time` and used in multiple other places, so moving `detect_link_tag_time` to the `Entry` class would cause issues around that.

Separating the tag utility methods out into their own `GeneratorTags` module instead of being part of either the `Generator` or `Entry` classes would be preferable, but is beyond the scope of this PR.

Reply:
The `newer_tag` values of name/link/time should be consistently handled in that order inside the `Entry` class. I have done some small refactoring to make this so.